internal/engine/executor.go

```diff
@@ -209,14 +208,55 @@ func (e *Executor) prepAndEvalEntityEvent(ctx context.Context, inf *entities.Ent
 
 	ectx := &EntityContext{
 		Project: Project{
-			ID: projectID,
+			ID: inf.ProjectID,
 		},
 		Provider: Provider{
 			Name: inf.Provider,
 		},
 	}
 
-	return e.evalEntityEvent(ctx, inf, ectx, cli)
+	// Channel the event based on the webhook action - create, update, delete, published, etc.
+	switch inf.ActionEvent {
+	case "deleted":
+		return e.deleteEntityEvent(ctx, inf, ectx)
```

Comment:
Uhm... I was thinking we'd have a reconciler dedicated towards delete events, and keep the executor just executing policy.

Reply:
That's a fair point. I started writing it, but then I felt it's not worth to add so much code just for this use case as it turned out it is not really useful for anything else yet other than repositories, i.e. artifacts and pull requests don't have events similar to `delete`. If you say it'd be better for future where we have different providers and such I can update it, no problem 👍

Reply:
Let's keep the executor just executing. Having a general deletion reconciler will pay off on the future.

Reply:
Alright, I'll rework the PR then 👍

Reply:
Updated the PR by adding a new reconciler and also moved queue topics to live under `events/constants.go` so it's more consistent.